cmake/FindFilesystem.cmake

```diff
@@ -124,6 +126,13 @@ cmake_push_check_state()
 
 set(CMAKE_REQUIRED_QUIET ${Filesystem_FIND_QUIETLY})
 
+# All of our tests required C++17 or later
+if(DEFINED CMAKE_CXX_STANDARD)
+    set(_prior_cmake_cxx_standard ${CMAKE_CXX_STANDARD})
+endif()
+
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Is there a reason to use C++17 here if ACTS as a whole requires C++20 these days?

Reply:
Not really, vector-of-bool's implentation uses this, I guess because that's when `std::filesystem` entered the standard.